### PR TITLE
RELENG-150 - Add mark-as-shipped task in Android-Components

### PR DESCRIPTION
**Pull request by JohanLorenzo**
This backports:
 * https://github.com/mozilla-mobile/fenix/pull/7306
 * https://github.com/mozilla-mobile/fenix/pull/10021
 * https://github.com/mozilla-mobile/fenix/pull/13257

to A-C.

No rush in landing this patch. This depends on https://github.com/mozilla-mobile/android-components/pull/9077 because of `post-beetmover` tasks and changes made in the `all_dep` loader. This also depends on https://github.com/mozilla-releng/shipit/pull/547 +  https://github.com/mozilla-mobile/android-components/pull/9132, because this task does not appear on any existing taskcluster graph. It will show up once we enable Android-Components on shipit.

My first testing with taskgraph-diff is promising 👍 I'll wait for the PRs above to get this tested end-to-end.

### CODE REVIEW

**Review by escapewindow — APPROVED** [review]
This looks good! I pointed out a few places where we've diverged from the staging fork and other merged PRs.

Also, we haven't landed the `_fill_treeherder_groups` changes and treeherder group names config changes from https://github.com/escapewindow/android-components/commit/aead9f6916f39d48dc5906f07a1d35e24a6db28e#diff-d011a51af6df810754a68ed3a9672eec9d430d4bd5b198b43605b5f43ab74be7 anywhere. Do we need that change? If so, we probably want to include it in this PR or in another PR.

r=me with the comments and above change ^.

File: taskcluster/ac_taskgraph/transforms/mark_as_shipped.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            }
+        )
+        yield task
+
+
+@transforms.add
+def make_task_description(config, jobs):
+    for job in jobs:
+        # TODO before landing: check the name of product on shipit (case sensitive?)
+        product = "Android-Components"
```
[the remark this answers is not in the document]
Comment (JohanLorenzo) [reply taskcluster/ac_taskgraph/transforms/mark_as_shipped.py]: Yup, I think prod will behave the same way. Fixed

File: taskcluster/ci/mark-as-shipped/kind.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    - ac_taskgraph.transforms.all_dep:transforms
+    - ac_taskgraph.transforms.mark_as_shipped:transforms
+    - taskgraph.transforms.task:transforms
+
+kind-dependencies:
+    # TODO before landing: Use post-beetmover once https://github.com/mozilla-mobile/android-components/pull/9077 lands
+    - post-signing
+
+# TODO before landing: Use post-beetmover once https://github.com/mozilla-mobile/android-components/pull/9077 lands
+primary-dependency: post-signing
```
[the remark this answers is not in the document]
Comment (JohanLorenzo) [reply taskcluster/ci/mark-as-shipped/kind.yml]: Thanks! 👍

File: taskcluster/ci/config.yml [hunk 3]
```diff
@@ -32,16 +32,26 @@ workers:
             implementation: docker-worker
             os: linux
             worker-type: b-linux-large
+        beetmover:
```
[the remark this answers is not in the document]
Comment (JohanLorenzo) [reply taskcluster/ci/config.yml:36]: Agreed! Done.

File: taskcluster/ci/mark-as-shipped/kind.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            default:
+                - project:releng:ship-it:action:mark-as-shipped
+                - project:releng:ship-it:server:staging
+    worker: {}
+
+    treeherder:
+        symbol: mark-shipped
+        platform: release/opt
+        kind: build
+        tier: 2
```
[the remark this answers is not in the document]
Comment (JohanLorenzo) [reply taskcluster/ci/mark-as-shipped/kind.yml:39]: Fine by me!

**Comment by JohanLorenzo** [comment]
> Also, we haven't landed the _fill_treeherder_groups changes and treeherder group names config changes

Good call! I just created https://github.com/mozilla-mobile/android-components/pull/9228 to fill this gap! 

Thanks a lot for catching all these discrepancies! 

> I'll wait for the PRs above to get this tested end-to-end.

Tested end-to-end at https://firefox-ci-tc.services.mozilla.com/tasks/Uf4EtWtdSQ-EQA4_FtJphQ. Each PR has landed. We're good to proceed!

**Pull request merged**